rest/config_test.go

```diff
@@ -440,3 +443,370 @@ func TestAutoImportEnabled(t *testing.T) {
+	// unsupported.stats_log_freq_secs
+	statsLogFrequencySecs := uint(9)
+	unsupported := &UnsupportedServerConfig{StatsLogFrequencySecs: &statsLogFrequencySecs}
+	sc := &ServerConfig{Unsupported: unsupported}
+	assert.Len(t, sc.validate(), 1)
+
+	// Valid configuration value for StatsLogFrequencySecs
+	statsLogFrequencySecs = uint(10)
+	unsupported = &UnsupportedServerConfig{StatsLogFrequencySecs: &statsLogFrequencySecs}
+	sc = &ServerConfig{Unsupported: unsupported}
+	assert.Len(t, sc.validate(), 0)
+
+	// Explicitly disabled
+	statsLogFrequencySecs = uint(0)
+	unsupported = &UnsupportedServerConfig{StatsLogFrequencySecs: &statsLogFrequencySecs}
+	sc = &ServerConfig{Unsupported: unsupported}
+	assert.Len(t, sc.validate(), 0)
+}
+
+func TestSetupAndValidateDatabases(t *testing.T) {
+	// No error will be returned if the server config itself is nil
+	var sc *ServerConfig
+	errs := sc.setupAndValidateDatabases()
+	assert.Nil(t, errs)
+
+	// Simulate  invalid control character in URL while validating and setting up databases;
+	server := "walrus:\n\r"
+	bc := &BucketConfig{Server: &server}
+	databases := make(DbConfigMap, 2)
+	databases["db1"] = &DbConfig{Name: "db1", BucketConfig: *bc}
+
+	sc = &ServerConfig{Databases: databases}
+	errs = sc.setupAndValidateDatabases()
+	assert.Len(t, errs, 1)
+	assert.Contains(t, errs[0].Error(), "invalid control character in URL")
+}
+
+func TestParseCommandLine(t *testing.T) {
+	func() { config = nil }()
+	var (
+		adminInterface     = "127.0.0.1:4985"
+		bucket             = "sync_gateway"
+		cacertpath         = "/etc/ssl/certs/ca.cert"
+		certpath           = "/etc/ssl/certs/client.pem"
+		configServer       = "http://127.0.0.1:4981/conf"
+		dbname             = "beer_sample"
+		defaultLogFilePath = "/var/log/sync_gateway"
+		deploymentID       = "DEPID100"
+		interfaceAddress   = "4984"
+		keypath            = "/etc/ssl/certs/key.pem"
+		log                = "Admin,Access,Auth,Bucket"
+		logFilePath        = "/var/log/sync_gateway"
+		pool               = "liverpool"
+	)
+	args := []string{
+		"sync_gateway",
+		"--adminInterface", adminInterface,
+		"--bucket", bucket,
+		"--cacertpath", cacertpath,
+		"--certpath", certpath,
+		"--configServer", configServer,
+		"--dbname", dbname,
+		"--defaultLogFilePath", defaultLogFilePath,
+		"--deploymentID", deploymentID,
+		"--interface", interfaceAddress,
+		"--keypath", keypath,
+		"--log", log,
+		"--logFilePath", logFilePath,
+		"--pool", pool,
+		"--pretty"}
+
+	err := ParseCommandLine(args, flag.ContinueOnError)
+	require.NoError(t, err, "Error reading config file")
+	conf := GetConfig()
+	assert.Equal(t, adminInterface, *conf.AdminInterface)
+	databases := conf.Databases
+	assert.Len(t, databases, 1)
+	assert.Equal(t, dbname, databases[dbname].Name)
+	assert.Equal(t, cacertpath, databases[dbname].CACertPath)
+	assert.Equal(t, certpath, databases[dbname].CertPath)
+	assert.Equal(t, keypath, databases[dbname].KeyPath)
```

Comment:
Should assert all of the command line parameters to guard against regressions.

Reply:
Asserted all of the command line parameters to guard against regressions.